src/peer_connection/peer_connection_internal.rs

```diff
@@ -1346,13 +1352,90 @@ impl PeerConnectionInternal {
 
     pub(super) async fn get_stats(&self, stats_id: String) -> StatsCollector {
         let collector = StatsCollector::new();
+        let collect_inbound_rtp_stats = {
+            let collector = &collector;
+
+            let fut = async move {
+                let transceivers = { self.rtp_transceivers.lock().await.clone() };
+
+                // TODO: There's a lot of await points here that could run concurrently with `futures::join_all`.
+                struct TrackInfo {
+                    ssrc: SSRC,
+                    mid: String,
+                    track_id: String,
+                    kind: &'static str,
+                }
+                let mut track_infos = vec![];
+                for transeiver in transceivers {
+                    let receiver = match transeiver.receiver().await {
+                        Some(r) => r,
+                        None => continue,
+                    };
+                    let mid = match transeiver.mid().await {
+                        m if !m.is_empty() => m,
+                        _ => continue,
+                    };
+
+                    let tracks = receiver.tracks().await;
+
+                    for track in tracks {
+                        let track_id = track.id().await;
+                        let kind = match track.kind() {
+                            RTPCodecType::Unspecified => continue,
+                            RTPCodecType::Audio => "audio",
+                            RTPCodecType::Video => "video",
+                        };
+
+                        track_infos.push(TrackInfo {
+                            ssrc: track.ssrc(),
+                            mid: mid.clone(),
+                            track_id,
+                            kind,
+                        });
+                    }
+                }
+
+                let readers = self
+                    .stats_interceptor
+                    .recv_stats_readers(track_infos.iter().map(|t| t.ssrc));
+
+                for (reader, info) in readers.into_iter().zip(track_infos) {
+                    let id = format!("RTCInboundRTP{}Stream_{}", capitalize(info.kind), info.ssrc);
```

Comment:
This attempts to match Chrome's ids for stats ids. @sax Did you try to do this for the other ids or did you establish some other convention?

Reply:
Where there was an id coming from a lower-level system I retained it, but where there was no id I tried to do the same thing.

Last I looked at different browsers was over a year ago, at which point it seemed like Safari was close (but not exactly the same) as Chrome, and Firefox did its own thing. So much has changed in the last year in terms of webrtc, however.

Reply:
Hmm, I guess I should review what the others do and establish a convention then